pkgs/top-level/dotnet-packages.nix

```diff
@@ -220,6 +220,37 @@ let self = dotnetPackages // overrides; dotnetPackages = with self; {
 
   # SOURCE PACKAGES
 
+  Boogie = buildDotnetPackage rec {
+    baseName = "Boogie";
```

Comment:
Hyphens in the version part is fine, the version will start at the first dash/digit pair. A demonstration:
```
nix-repl> builtins.parseDrvName "Boogie-unstable-2017-01-03"
{ name = "Boogie-unstable"; version = "2017-01-03"; }

nix-repl> builtins.parseDrvName "Boogie-unstable-2017-01-03-b1"
{ name = "Boogie-unstable"; version = "2017-01-03-b1"; }

nix-repl> builtins.parseDrvName "Boogie-unstable-b1-2017-01-03"
{ name = "Boogie-unstable-b1"; version = "2017-01-03"; }
```
Version comparison will also work OK with ISO dates:
```
nix-repl> builtins.compareVersions "2017-01-03" "2017-02-03"
-1

nix-repl> builtins.compareVersions "2017-01-03" "2017-01-03"
0

nix-repl> builtins.compareVersions "2017-01-03" "2016-12-03"
1

nix-repl> # Even supports 0th month
          builtins.compareVersions "2017-01-03" "2017-00-03"
1
```
So such versions will work as expected and indeed is part of the [policy](https://nixos.org/nixpkgs/manual/#sec-package-naming):
> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".

It took me a while to come to terms with the YYYY-MM-DD form of versions but now I am a champion :-)